File: Easy/844. Backspace String Compare/backspaceCompare.cpp
class Solution {
public:
    bool backspaceCompare(string S, string T) {
        return check(S) == check(T);
    }
private:
    string check(string &s){
        string tmp = "";
        for(char c : s){
            if(c == '#'){
                if(!tmp.empty()){
                    tmp.pop_back();
                }
            }else{
                tmp.push_back(c);
            }
        }
        return tmp;
    }
};

File: Easy/844. Backspace String Compare/backspaceCompare.java
class Solution {
  public boolean backspaceCompare(String S, String T) {
      return check(S).equals(check(T));
  }
  private String check(String s) {
      StringBuilder sb = new StringBuilder();
      
      for (char c : s.toCharArray()) {
        if (c == '#') {
          if (sb.length() > 0)
            sb.deleteCharAt(sb.length() - 1);
          }else{
            sb.append(c);
          }
        }
      return sb.toString();
  }
}